### PR TITLE
Minimize host-verify package without separating oehost

### LINKED ISSUE #2263: Allow building only hostverify library and with minimal build dependencies
OE provides roughly two parts: a) multiple libraries to develop and manage enclaves (target: platform that runs enclaves), and b) the hostverify library to ensure secure communication with remote enclaves (target: any platform, e.g. lightweight clients without enclave support).

Building clients that use the hostverify library currently is unnecessarily complicated. In many cases the provided Ubuntu packages are not enough, either because a different distribution needs to be supported or because there are policies to build all dependencies from source. Building from source is hard because OE's CMake scripts don't allow to build *only* the hostverify library and hence many unneeded build dependencies are required, including Intel's SGX libraries and an up-to-date version of the OCaml compiler.

IMO easily building the hostverify library from source should have first-class support.

### PR DESCRIPTION
**Pull request by ryanhsu19**
This is a workaround based on pull request #3123 and issues #3160 and #3181 so that we do not have to revise any logic or testing flow for now.

This change makes it possible to use `cpack` to create a usable `open-enclave-hostverify` package with only necessary header files and targets, including `openenclave::oehost` ~~and `openenclave::oesign`~~ which ~~are~~ is unable to separate; that is, we can build the `host-verify` sample without any modification of files installed from `open-enclave-hostverify`.

This pull request fixes #2263 partially, since it still includes targets `openenclave::oehost` ~~and `openenclave::oesign`~~.

Signed-off-by: Ryan Hsu <ryhsu@microsoft.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,13 +118,7 @@ else ()
...
 
 if (OE_SGX)
-  if (WIN32)
-    # Building enclaves on windows is on by default but can be disabled for enclaves pre-compiled under linux
-    option(BUILD_ENCLAVES "Build ELF enclaves" ON)
-  else ()
-    set(BUILD_ENCLAVES ON)
-  endif ()
-
+  set(BUILD_ENCLAVES ON)
```
Comment (mingweishih) [inline CMakeLists.txt:121]: On Windows, we still want that `BUILD_ENCLAVES` can be overwritten. Can you revert the changes?

File: samples/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
-if (HAS_QUOTE_PROVIDER)
+if (BUILD_ENCLAVES AND HAS_QUOTE_PROVIDER)
```
Comment (mingweishih) [inline samples/CMakeLists.txt:4]: Is there change in this PR that allows sample to be built when `BUILD_ENCLAVES` is `OFF`?

File: samples/test-samples.cmake [hunk 3]
```diff
@@ -6,7 +6,12 @@
 #
 #     cmake -DHAS_QUOTE_PROVIDER=ON -DSOURCE_DIR=~/openenclave -DBUILD_DIR=~/openenclave/build -DPREFIX_DIR=/opt/openenclave -P ~/openenclave/samples/test-samples.cmake
 
-set(SAMPLES_LIST helloworld file-encryptor switchless host_verify)
+# Set SAMPLES_LIST so that helloworld becomes the first if BUILD_ENCLAVES=ON.
+if (BUILD_ENCLAVES)
```
Comment (mingweishih) [inline samples/test-samples.cmake:10]: Same here, see the question above.

File: samples/test-samples.cmake [hunk 4]
```diff
@@ -21,13 +26,15 @@ else ()
...
   # they can run even if they weren't built against SGX, because in
   # that cause they directly interface with the AESM service.
-  list(APPEND SAMPLES_LIST data-sealing local_attestation)
-
-  # These tests can only run with SGX-FLC, meaning they were built
-  # against SGX.
-  if (HAS_QUOTE_PROVIDER)
-    list(APPEND SAMPLES_LIST remote_attestation)
-    list(APPEND SAMPLES_LIST attested_tls)
+  if (BUILD_ENCLAVES)
```
Comment (mingweishih) [inline samples/test-samples.cmake:29]: Same here, see the question above.

**Comment by mingweishih** [comment]
bors try

File: samples/CMakeLists.txt (hunk 2 above)
Comment (ryanhsu19) [reply samples/CMakeLists.txt:4]: Just modified `CMakeLists.txt` in the root folder so that `host_verify` sample can always be built regardless of the value of `BUILD_ENCLAVES`.

**Comment by ryanhsu19** [comment]
Update again to exclude `oesign` from host-verify package.
I use the following command to generate the host-verify package on Ubuntu 18.04 LTS:

 - `cmake -DCOMPONENT=OEHOSTVERIFY -DCMAKE_INSTALL_PREFIX=/opt/openenclave ..`

 - `cpack -G DEB -DCPACK_DEB_COMPONENT_INSTALL=ON -DCPACK_COMPONENTS_ALL=OEHOSTVERIFY`

**Comment by ryanhsu19** [comment]
bors try

File: tools/oesign/CMakeLists.txt [hunk 5]
```diff
@@ -22,25 +22,27 @@ if (NOT HAVE_GETOPT_LONG)
   list(APPEND SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/getopt_long.c)
 endif ()
 
-add_executable(oesign ${SOURCES})
+if (NOT COMPONENT MATCHES OEHOSTVERIFY)
```
Comment (mingweishih) [inline tools/oesign/CMakeLists.txt:25]: Is this check needed? We already have
```
if (NOT COMPONENT MATCHES OEHOSTVERFIY)
  add_subdirectory(tools)
endif ()
```
in the `CMakeLists.txt`

Comment (ryanhsu19) [reply tools/oesign/CMakeLists.txt:25]: Right, I just tried, but it seems that there are other places where `tools/oesign` can be accessed.

**Comment by mingweishih** [comment]
Please sign off and squash the commits so we can merge. Thanks!

**Comment by ryanhsu19** [comment]
bors try
